### DISCUSSION TITLE
The web side exposes server information.

### DISCUSSION THREAD

**Discussion by LaoJiuYes** [ente-io/ente#2578] [Q&A]
![image](https://github.com/user-attachments/assets/f8a31d7f-0d23-4dd7-9ffa-aef078d35329)
Does this one affect safety? How do I turn it off, I looked at the generated web js file and there are three places where it is there, and when I turn it off, the server connection is broken.

**Comment by mnvr** [comment]
This does not affect safety. This information is trivially available to anyone by opening the dev tools in the browser and seeing where the app makes connections when it does a network request.

This was added so that it is easy to see for people who are self hosting that everything is configured properly, and that they will actually connect to their own self hosted instance instead of Ente's servers.

If it really bothers you, you can remove it by returning `undefined` from `customAPIHost` - https://github.com/ente-io/ente/blob/main/web/packages/base/origins.ts#L60

But be aware that that workaround might break in the future - currently the `customAPIHost` is only used for indicating in the UI, but in the future it might also get used for functional changes.

**Reply by LaoJiuYes** [reply]
> This does not affect safety. This information is trivially available to anyone by opening the dev tools in the browser and seeing where the app makes connections when it does a network request.
> 
> This was added so that it is easy to see for people who are self hosting that everything is configured properly, and that they will actually connect to their own self hosted instance instead of Ente's servers.
> 
> If it really bothers you, you can remove it by returning `undefined` from `customAPIHost` - https://github.com/ente-io/ente/blob/main/web/packages/base/origins.ts#L60
> 
> But be aware that that workaround might break in the future - currently the `customAPIHost` is only used for indicating in the UI, but in the future it might also get used for functional changes.

Thank you for your reply.
I understand what you said, thanks for your explanation. The reason why I ask this question is because in China, our home broadband does not have a fixed IP (also known as “public IP”), and we all use the carrier's NAT intranet IP. if I am outside my home, I can't access my home network directly.
I have a NAS at home, and then deploy ente photos server and front end, I need to buy a cloud server for NAT intranet penetration, so that I can access my NAS outside my home, and upload my photos to ente photos at any time. I'm afraid that others will also access my website, which will expose some information about my server.
The ente photos interface is beautiful and the experience is great. Thanks for making it open source. I offer my respect.
